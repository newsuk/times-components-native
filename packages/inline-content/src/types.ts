import { ParagraphContent } from "@times-components-native/types";

export type Line = {
  text: string;
};
export type TextLayoutEvent = { lines: Line[] };
export type ChunkContents = ParagraphContent[];

export interface ContentParameters {
  contentWidth: number;
  contentHeight: number;
  contentLineHeight: number;
  itemWidth: number;
}

export type SkeletonProps = Record<string, unknown>;

interface BaseInlineContentProps {
  adConfig: Record<string, unknown>;
  defaultFont: { lineHeight: number };
  display: string;
  height: number;
  inlineContent: ParagraphContent[];
  narrowContent: boolean;
  originalName: string;
  skeletonProps: SkeletonProps;
  slotName: string;
  width: number;
}

export interface InlineArticleImageProps extends BaseInlineContentProps {
  caption: string;
  credits: string;
  imageIndex: number;
  narrowContent: boolean;
  onImagePress: () => void;
  ratio: string;
  relativeWidth: number;
  relativeHeight: number;
  relativeHorizontalOffset: number;
  relativeVerticalOffset: number;
  url: string;
}

export interface InlinePullQuoteProps extends BaseInlineContentProps {
  caption: { name: string; text: string; twitter: string };
  children: [{ string: string }];
  onTwitterLinkPress: () => void;
}

export type InlineContentProps =
  | BaseInlineContentProps
  | InlineArticleImageProps
  | InlinePullQuoteProps;

export type ArticleImageProps = {
  captionOptions: {
    caption: string;
    credits: string;
  };
  onImagePress: () => void;
  images: [];
  imageOptions: {
    display: string;
    ratio: string;
    index: number;
    uri: string;
    relativeWidth: number;
    relativeHeight: number;
    relativeHorizontalOffset: number;
    relativeVerticalOffset: number;
    narrowContent: boolean;
  };
  originalName: string;
};

export type PullQuoteProps = {
  caption: string;
  children: [{ string: string }];
  onTwitterLinkPress: () => void;
  originalName: string;
  text: string;
  twitter: string;
  width: number;
};

export type InlineItemProps = ArticleImageProps | PullQuoteProps;
