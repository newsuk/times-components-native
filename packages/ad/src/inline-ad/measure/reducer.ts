import { Reducer } from "react";
import { ContentMeasurements, Line } from "../types";

export const initialState: ContentMeasurements = {
  contents: {
    lines: {},
    heights: {},
  },
};

interface SetInlineContentHeightAction {
  type: "SET_INLINE_CONTENT_HEIGHT";
  id: string;
  payload: number;
}

interface SetInlineContentLinesAction {
  type: "SET_INLINE_CONTENT_LINES";
  id: string;
  payload: Line[];
}

export type Action = SetInlineContentHeightAction | SetInlineContentLinesAction;

export const reducer: Reducer<ContentMeasurements, Action> = (
  state,
  action,
) => {
  switch (action.type) {
    case "SET_INLINE_CONTENT_HEIGHT":
      return {
        ...state,
        contents: {
          ...state.contents,
          heights: { ...state.contents.heights, [action.id]: action.payload },
        },
      };
    case "SET_INLINE_CONTENT_LINES":
      return {
        ...state,
        contents: {
          ...state.contents,
          lines: { ...state.contents.lines, [action.id]: action.payload },
        },
      };
    default:
      return state;
  }
};
