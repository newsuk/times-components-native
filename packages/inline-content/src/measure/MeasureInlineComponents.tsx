import React, { memo } from "react";
import { View } from "react-native";

import ArticleImage from "@times-components-native/article-image";
import renderTrees from "@times-components-native/markup-forest";
import getRenderers from "@times-components-native/article-skeleton/src/article-body/article-body-row.js";
import { ParagraphContent } from "@times-components-native/types";
import { useInlineMeasurementDispatchContext } from "./InlineMeasurementDispatchContext";
import { ArticleImageProps, SkeletonProps } from "../types";

interface Props {
  content: ParagraphContent;
  skeletonProps: SkeletonProps;
}

export const MeasureContent: React.FC<Props> = memo(
  ({ content, skeletonProps }) => {
    const dispatch = useInlineMeasurementDispatchContext();

    const inlineContent = {
      ...content,
      attributes: { ...content.attributes, inline: true },
    };

    const { data, isTablet, narrowContent, scale } = skeletonProps;

    return (
      <View
        onLayout={(event) => {
          const height = Math.floor(event.nativeEvent.layout.height);
          dispatch({
            type: "SET_INLINE_CONTENT_HEIGHT",
            id: inlineContent.id!,
            payload: height,
          });
        }}
      >
        {renderTrees(
          [inlineContent],
          getRenderers({
            data,
            interactiveConfig: {},
            onLinkPress: null,
            onTwitterLinkPress: null,
            onVideoPress: null,
            onImagePress: null,
            isTablet,
            adConfig: {},
            images: [],
            dropcapsDisabled: true,
            dropCapFont: "dropCap",
            scale,
            analyticsStream: null,
            narrowContent,
            onParagraphTextLayout: (event: any) => {
              const lines = event.nativeEvent.lines;
              dispatch({
                type: "SET_INLINE_CONTENT_LINES",
                id: inlineContent.id!,
                payload: lines,
              });
            },
          }),
        )}
      </View>
    );
  },
);

export const MeasureItem: React.FC<{
  itemProps: ArticleImageProps;
  width: number;
}> = memo(({ itemProps, width }) => {
  const dispatch = useInlineMeasurementDispatchContext();

  if (!itemProps || !width)
    dispatch({
      type: "SET_INLINE_ITEM_HEIGHT",
      height: 0,
    });

  return (
    <View
      style={{ width }}
      onLayout={(event) => {
        const height = event.nativeEvent.layout.height;
        dispatch({
          type: "SET_INLINE_ITEM_HEIGHT",
          height,
        });
      }}
    >
      <ArticleImage {...itemProps} />
    </View>
  );
});
