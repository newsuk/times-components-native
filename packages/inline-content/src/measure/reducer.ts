import { Reducer } from "react";

import { Measurements } from "@times-components-native/types";
import { Line } from "../types";

export const initialState: Measurements = {
  contents: {
    lines: {},
    heights: {},
  },
  itemHeight: null,
};

interface SetInlineContentHeightAction {
  type: "SET_INLINE_CONTENT_HEIGHT";
  id: string;
  payload: number;
}

interface SetInlineContentLinesAction {
  type: "SET_INLINE_CONTENT_LINES";
  id: string;
  payload: Line[];
}

interface SetInlineItemHeightAction {
  type: "SET_INLINE_ITEM_HEIGHT";
  height: number;
}

export type Action =
  | SetInlineContentHeightAction
  | SetInlineContentLinesAction
  | SetInlineItemHeightAction;

export const reducer: Reducer<Measurements, Action> = (state, action) => {
  switch (action.type) {
    case "SET_INLINE_CONTENT_HEIGHT":
      return {
        ...state,
        contents: {
          ...state.contents,
          heights: { ...state.contents.heights, [action.id]: action.payload },
        },
      };
    case "SET_INLINE_CONTENT_LINES":
      return {
        ...state,
        contents: {
          ...state.contents,
          lines: { ...state.contents.lines, [action.id]: action.payload },
        },
      };
    case "SET_INLINE_ITEM_HEIGHT":
      return {
        ...state,
        itemHeight: action.height,
      };
    default:
      return state;
  }
};
