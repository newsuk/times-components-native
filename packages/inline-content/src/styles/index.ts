import { StyleSheet, ViewStyle } from "react-native";

import { spacing, tabletWidth } from "@times-components-native/styleguide";

interface Styles {
  container: ViewStyle;
  inlineItemContainer: ViewStyle;
}

const styles: Styles = {
  container: {
    flex: 1,
    flexDirection: "row",
    width: tabletWidth,
  },
  inlineItemContainer: {
    marginLeft: spacing(2),
    marginBottom: spacing(2),
  },
};

export default StyleSheet.create<Styles>(styles);
