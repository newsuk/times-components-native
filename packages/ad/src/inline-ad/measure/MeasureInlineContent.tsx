import React, { useReducer } from "react";
import { ScrollView, StyleSheet, View } from "react-native";

import { MeasureContent } from "./MeasureInlineComponents";
import { InlineMeasurementDispatch } from "./InlineMeasurementDispatchContext";
import { initialState, reducer } from "./reducer";
import { Measurements, ParagraphContent } from "@times-components-native/types";
import { ContentParameters, Line, SkeletonProps } from "../types";

interface Props {
  content: ParagraphContent[];
  contentParameters: ContentParameters;
  renderMeasuredContents: (contentMeasurements: Measurements) => any;
  skeletonProps: SkeletonProps;
}

const allContentMeasured = (
  contentToMeasure: ParagraphContent[],
  heights: { [key: string]: number },
  lines: { [key: string]: Line[] },
) => {
  return (
    contentToMeasure.filter((c) => heights[c.id!] && lines[c.id!]).length ===
    contentToMeasure.length
  );
};

export const InnerMeasureInlineContent: React.FC<
  Props & { measurementState: Measurements }
> = ({
  content,
  contentParameters,
  skeletonProps,
  renderMeasuredContents,
  measurementState,
}) => {
  const {
    contents: { lines, heights },
  } = measurementState;
  if (allContentMeasured(content, heights, lines)) {
    return renderMeasuredContents(measurementState);
  }

  return (
    <ScrollView style={styles.renderOffscreen}>
      <View
        style={{
          width: contentParameters.contentWidth,
        }}
      >
        {content.map((contentItem) => (
          <MeasureContent
            key={`InlineContentMeasuringView:${contentItem.id}`}
            skeletonProps={skeletonProps}
            content={contentItem}
          />
        ))}
      </View>
    </ScrollView>
  );
};

/* istanbul ignore next */
export const MeasureInlineContent: React.FC<Props> = (props) => {
  const [measurementState, measurementDispatch] = useReducer(
    reducer,
    initialState,
  );

  return (
    <InlineMeasurementDispatch.Provider value={measurementDispatch}>
      <InnerMeasureInlineContent
        {...props}
        measurementState={measurementState}
      />
    </InlineMeasurementDispatch.Provider>
  );
};

const styles = StyleSheet.create({
  renderOffscreen: {
    transform: [{ translateX: -1000 }],
  },
});
