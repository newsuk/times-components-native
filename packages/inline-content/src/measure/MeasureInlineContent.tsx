import React, { useReducer } from "react";
import { ScrollView, StyleSheet, View } from "react-native";

import { Measurements, ParagraphContent } from "@times-components-native/types";
import { InlineMeasurementDispatch } from "./InlineMeasurementDispatchContext";
import { MeasureItem, MeasureContent } from "./MeasureInlineComponents";
import { initialState, reducer } from "./reducer";
import {
  ContentParameters,
  InlineItemProps,
  Line,
  SkeletonProps,
} from "../types";

interface Props {
  content: ParagraphContent[];
  contentParameters: ContentParameters;
  itemProps: InlineItemProps;
  renderMeasuredContents: (contentMeasurements: Measurements) => any;
  skeletonProps: SkeletonProps;
}

const allContentMeasured = (
  contentToMeasure: ParagraphContent[],
  heights: { [key: string]: number },
  lines: { [key: string]: Line[] },
) => {
  return (
    contentToMeasure.filter((c) => heights[c.id!] && lines[c.id!]).length ===
    contentToMeasure.length
  );
};

export const InnerMeasureInlineContent: React.FC<
  Props & { measurementState: Measurements }
> = ({
  content,
  contentParameters,
  itemProps,
  measurementState,
  renderMeasuredContents,
  skeletonProps,
}) => {
  const {
    contents: { lines, heights },
  } = measurementState;
  if (
    allContentMeasured(content, heights, lines) &&
    (!itemProps || measurementState.itemHeight !== null)
  ) {
    return renderMeasuredContents(measurementState);
  }

  return (
    <ScrollView style={styles.renderOffscreen}>
      {itemProps && (
        <MeasureItem
          itemProps={itemProps}
          width={contentParameters.itemWidth}
        />
      )}
      <View
        style={{
          width: contentParameters.contentWidth,
        }}
      >
        {content.map((contentItem) => (
          <MeasureContent
            key={`InlineContentMeasuringView:${contentItem.id}`}
            skeletonProps={skeletonProps}
            content={contentItem}
          />
        ))}
      </View>
    </ScrollView>
  );
};

/* istanbul ignore next */
export const MeasureInlineContent: React.FC<Props> = (props) => {
  const [measurementState, measurementDispatch] = useReducer(
    reducer,
    initialState,
  );

  return (
    <InlineMeasurementDispatch.Provider value={measurementDispatch}>
      <InnerMeasureInlineContent
        {...props}
        measurementState={measurementState}
      />
    </InlineMeasurementDispatch.Provider>
  );
};

const styles = StyleSheet.create({
  renderOffscreen: {
    transform: [{ translateX: -1000 }],
  },
});
