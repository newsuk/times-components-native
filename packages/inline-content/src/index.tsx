import React, { useCallback, FC } from "react";
import { View, Text } from "react-native";

import { ParagraphContent } from "@times-components-native/types";
import {
  colours,
  spacing,
  tabletWidth,
  getNarrowArticleBreakpoint,
} from "@times-components-native/styleguide";
import { render } from "@times-components-native/markup-forest";
import {
  renderers,
  Gutter,
  ErrorBoundary,
} from "@times-components-native/article-skeleton";
import { useResponsiveContext } from "@times-components-native/responsive";
import Context from "@times-components-native/context";
import ArticleImage from "@times-components-native/article-image";
import PullQuote from "@times-components-native/pull-quote";

import { chunkInlineContent } from "./utils/chunkInlineContent";
import { MeasureInlineContent } from "./measure/MeasureInlineContent";

import styles from "./styles";
import { ArticleImageProps, SkeletonProps } from "./types";

const assignWithId = (height: number) => (
  content: ParagraphContent,
  idx: number,
): ParagraphContent => {
  return {
    ...content,
    id: `${idx}-${height}`, //suffixing the height ensures that we re-measure the content if the orientation changes - and that we don't unnecessarily re-measure if orientation changes back
  };
};

export const renderItemComponent = (itemProps) => {
  const { children, originalName, width } = itemProps;
  if (originalName === "image") {
    return <ArticleImage {...itemProps} />;
  }
  if (originalName === "pullQuote") {
    const itemContent = children?.[0]?.string;
    return (
      <Context.Consumer>
        {({
          theme: { pullQuoteFont, sectionColour = colours.section.default },
        }) => (
          <View style={{ width }}>
            <PullQuote
              {...itemProps}
              font={pullQuoteFont}
              quoteColour={sectionColour}
            >
              {itemContent}
            </PullQuote>
          </View>
        )}
      </Context.Consumer>
    );
  }
};

interface Props {
  adConfig: Record<string, unknown>;
  defaultFont: { lineHeight: number };
  display: string;
  height: number;
  inlineContent: ParagraphContent[];
  originalName: string;
  skeletonProps: SkeletonProps;
  slotName: string;
  width: number;
}

const InlineContent: FC<Props> = (props) => {
  const {
    defaultFont,
    inlineContent,
    narrowContent,
    originalName,
    skeletonProps,
  } = props;

  const { windowWidth } = useResponsiveContext();

  const renderChild = render(
    // @ts-ignore
    renderers({ dropcapsDisabled: true, ...skeletonProps }),
  );

  const Child = useCallback(({ item, index }, inline = false) => {
    item.attributes = { ...item.attributes, inline };

    return (
      <Gutter style={{ overflow: "hidden" }}>
        <ErrorBoundary>
          {renderChild(item, index.toString(), index)}
        </ErrorBoundary>
      </Gutter>
    );
  }, []);

  const renderItem = (inline: boolean) => (
    item: ParagraphContent,
    index: number,
  ) => Child({ item, index }, inline);

  const { lineHeight } = defaultFont;

  const availableWidth = Math.min(
    windowWidth,
    narrowContent
      ? getNarrowArticleBreakpoint(windowWidth).content
      : tabletWidth,
  );

  const inlineItemWidth = availableWidth * 0.35;
  const inlineContentWidth = availableWidth - inlineItemWidth;

  let itemProps: ArticleImageProps;

  console.log("OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO", originalName);

  if (originalName === "image") {
    const {
      caption,
      credits,
      display,
      imageIndex,
      // narrowContent,
      onImagePress,
      ratio,
      relativeWidth,
      relativeHeight,
      relativeHorizontalOffset,
      relativeVerticalOffset,
      url,
    } = props;

    itemProps = {
      captionOptions: {
        caption,
        credits,
      },
      onImagePress,
      originalName,
      images: [],
      imageOptions: {
        display,
        ratio,
        index: imageIndex,
        uri: url,
        relativeWidth,
        relativeHeight,
        relativeHorizontalOffset,
        relativeVerticalOffset,
        narrowContent,
      },
    };
  }

  // if (originalName === "pullQuote") {
  const {
    caption: { name, text, twitter },
    children,
    onTwitterLinkPress,
  } = props;

  itemProps = {
    caption: name,
    children,
    onTwitterLinkPress,
    originalName,
    text,
    twitter,
    width: inlineItemWidth,
  };
  // }

  // const [ratioWidth, ratioHeight] = ratio.split(":");
  // const aspectRatio = ratioWidth / ratioHeight;
  // const inlineItemHeight = inlineItemWidth / aspectRatio;
  // const inlineContentHeight = inlineItemHeight + spacing(2); // <<<<<<<<<<<<<<<< TODO!!!!!!!!!!!!!!

  // const adHeaderHeight = spacing(4);
  // const adHorizontalSpacing = 21;
  // const adMarginBottom = spacing(2);
  // const adContainerHeight = height + adHeaderHeight;
  // const adContainerHeightPlusMargin = adContainerHeight + adMarginBottom;
  // const adContainerWidth = width + adHorizontalSpacing;
  // const contentWidth = tabletWidth - adContainerWidth;

  // const contentHeight = adContainerHeightPlusMargin;

  const paragraphs = inlineContent
    .filter((c) => c.name === "paragraph")
    .map(assignWithId(inlineContentWidth));

  const contentParameters = {
    contentWidth: inlineContentWidth,
    // contentHeight: inlineContentHeight,
    contentHeight: 0,
    contentLineHeight: lineHeight,
    itemWidth: inlineItemWidth,
  };

  return (
    <MeasureInlineContent
      content={paragraphs}
      contentParameters={contentParameters}
      itemProps={itemProps}
      skeletonProps={skeletonProps}
      renderMeasuredContents={(contentMeasurements) => {
        const { chunks, currentInlineContentHeight } = chunkInlineContent(
          paragraphs,
          contentMeasurements,
          contentParameters,
        );

        // const itemHeight =
        //   contentMeasurements.itemHeight || inlineContentHeight || 0;
        const itemHeight = contentMeasurements.itemHeight || 0;

        const requiredInlineContentHeight = Math.max(
          currentInlineContentHeight,
          itemHeight,
        );

        const chunkedInlineContent = chunks[0] || [];
        const chunkedOverflowContent = chunks[1] || [];

        return (
          <>
            <View
              style={[
                styles.container,
                { height: requiredInlineContentHeight },
              ]}
            >
              <View
                style={[
                  narrowContent
                    ? styles.inlineItemNarrowContainer
                    : styles.inlineItemContainer,
                  { width: inlineItemWidth, height: itemHeight },
                ]}
              >
                {renderItemComponent(itemProps)}
                {/* <ItemComponent {...itemProps} /> */}
                {/* {Component} */}
              </View>
              <View
                style={{
                  height: requiredInlineContentHeight,
                  width: inlineContentWidth,
                }}
              >
                {chunkedInlineContent.map(renderItem(true))}
              </View>
            </View>
            {chunkedOverflowContent.map(renderItem(false))}
          </>
        );
      }}
    />
  );
};

export default InlineContent;
