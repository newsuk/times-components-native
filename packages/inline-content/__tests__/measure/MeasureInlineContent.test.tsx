import React from "react";
import { View, Text } from "react-native";
import TestRenderer from "react-test-renderer";

import { InnerMeasureInlineContent } from "../../src/measure/MeasureInlineContent";
import { initialState } from "../../src/measure/reducer";
import {
  ArticleContent,
  ParagraphContent,
} from "@times-components-native/types";
import { ContentParameters } from "../../src/types";

jest.mock("../../src/measure/MeasureInlineComponents", () => ({
  MeasureContent: "MeasureContent",
  MeasureItem: "MeasureItem",
}));

const createParagraphWithText = (
  text: string,
  testData: Partial<ParagraphContent> = {},
): ParagraphContent => ({
  id: "some-paragraph-id",
  name: "paragraph",
  children: [{ name: "text", children: [], attributes: { value: text } }],
  ...testData,
});

const paragraphA: ArticleContent = createParagraphWithText("line1", {
  id: "p1",
});

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const CaptureContents = ({ contentMeasurements }: any) => (
  <View>
    <Text>{JSON.stringify(contentMeasurements)}</Text>
  </View>
);

const contentParameters: ContentParameters = {
  contentHeight: 2000,
  contentWidth: 429,
  contentLineHeight: 26,
  itemWidth: 231,
};

describe("MeasureInlineContent", () => {
  it("renders content to be measured", () => {
    const renderer = TestRenderer.create(
      <InnerMeasureInlineContent
        contentParameters={contentParameters}
        content={[paragraphA]}
        renderMeasuredContents={() => null}
        skeletonProps={{ some: "props" }}
        measurementState={initialState}
        itemProps={undefined}
      />,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("outputs rendered content once everything has been measured", () => {
    const renderer = TestRenderer.create(
      <InnerMeasureInlineContent
        contentParameters={contentParameters}
        content={[paragraphA]}
        renderMeasuredContents={(contentMeasurements) => (
          <CaptureContents contentMeasurements={contentMeasurements} />
        )}
        skeletonProps={{ some: "props" }}
        measurementState={{
          ...initialState,
          contents: {
            lines: { p1: [{ text: "line1" }] },
            heights: { p1: 20 },
          },
        }}
        itemProps={undefined}
      />,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("renders content with inline item to be measured", () => {
    const renderer = TestRenderer.create(
      <InnerMeasureInlineContent
        contentParameters={contentParameters}
        content={[paragraphA]}
        renderMeasuredContents={() => null}
        skeletonProps={{ some: "props" }}
        measurementState={initialState}
        itemProps={{
          caption: "some caption",
          children: [{ string: "some pull quote text" }],
          onTwitterLinkPress: () => {
            return;
          },
          originalName: "pullQuote",
          text: "some text",
          twitter: "some twitter text",
          width: 231,
        }}
      />,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("outputs rendered content with inline item once everything has been measured", () => {
    const renderer = TestRenderer.create(
      <InnerMeasureInlineContent
        contentParameters={contentParameters}
        content={[paragraphA]}
        renderMeasuredContents={(contentMeasurements) => (
          <CaptureContents contentMeasurements={contentMeasurements} />
        )}
        skeletonProps={{ some: "props" }}
        measurementState={{
          contents: {
            lines: { p1: [{ text: "line1" }] },
            heights: { p1: 20 },
          },
          itemHeight: 300,
        }}
        itemProps={{
          caption: "some caption",
          children: [{ string: "some pull quote text" }],
          onTwitterLinkPress: () => {
            return;
          },
          originalName: "pullQuote",
          text: "some text",
          twitter: "some twitter text",
          width: 429,
        }}
      />,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
