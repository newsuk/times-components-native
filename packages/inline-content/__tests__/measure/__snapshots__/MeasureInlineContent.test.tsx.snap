// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeasureInlineContent outputs rendered content once everything has been measured 1`] = `
<View>
  <Text>
    {"contents":{"lines":{"p1":[{"text":"line1"}]},"heights":{"p1":20}},"itemHeight":null}
  </Text>
</View>
`;

exports[`MeasureInlineContent outputs rendered content with inline item once everything has been measured 1`] = `
<View>
  <Text>
    {"contents":{"lines":{"p1":[{"text":"line1"}]},"heights":{"p1":20}},"itemHeight":300}
  </Text>
</View>
`;

exports[`MeasureInlineContent renders content to be measured 1`] = `
<RCTScrollView
  style={
    Object {
      "transform": Array [
        Object {
          "translateX": -1000,
        },
      ],
    }
  }
>
  <View>
    <View
      style={
        Object {
          "width": 429,
        }
      }
    >
      <MeasureContent
        content={
          Object {
            "children": Array [
              Object {
                "attributes": Object {
                  "value": "line1",
                },
                "children": Array [],
                "name": "text",
              },
            ],
            "id": "p1",
            "name": "paragraph",
          }
        }
        skeletonProps={
          Object {
            "some": "props",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`MeasureInlineContent renders content with inline item to be measured 1`] = `
<RCTScrollView
  style={
    Object {
      "transform": Array [
        Object {
          "translateX": -1000,
        },
      ],
    }
  }
>
  <View>
    <MeasureItem
      itemProps={
        Object {
          "caption": "some caption",
          "children": Array [
            Object {
              "string": "some pull quote text",
            },
          ],
          "onTwitterLinkPress": [Function],
          "originalName": "pullQuote",
          "text": "some text",
          "twitter": "some twitter text",
          "width": 231,
        }
      }
      width={231}
    />
    <View
      style={
        Object {
          "width": 429,
        }
      }
    >
      <MeasureContent
        content={
          Object {
            "children": Array [
              Object {
                "attributes": Object {
                  "value": "line1",
                },
                "children": Array [],
                "name": "text",
              },
            ],
            "id": "p1",
            "name": "paragraph",
          }
        }
        skeletonProps={
          Object {
            "some": "props",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
